canonicalwebteam/http/__init__.py

```diff
@@ -100,24 +103,43 @@ class UncachedSession(BaseSession, requests.Session):
     pass
 
 
-class CachedSession(BaseSession, requests_cache.CachedSession):
+class CachedSession(BaseSession, Session):
     """
-    A session object for making HTTP requests with cached responses.
-
-    Responses for an identical request will be naively returned from the
-    cache if the cached copy if less than "expire_after" seconds old.
+    A session object that implements CacheControl functionality with
+    our default settings from BaseSession.
+
+    The session respects cache control headers to manage the caching strategy
+    and saves responses in a file or redis cache.
+    If CacheControl headers are not provided a custom duration time
+    for caching can be passed as a fallback strategy.
+
+    Keyword arguments:
+    redis_cache_pool -- Port of your Redis instance
+    file_cache_name -- Name for the file to store cache on
+    fallback_cache_duration -- Duration in seconds for fallback caching
+        retention when no CacheControl headers are set
     """
 
     def __init__(
         self,
-        backend='sqlite', expire_after=5, include_get_headers=True,
+        redis_cache_pool=None,
+        fallback_cache_duration=5,
+        file_cache_name=".webcache",
         *args,
         **kwargs
     ):
+        super(BaseSession, self).__init__(*args, **kwargs)
+
+        heuristic = ExpiresAfter(seconds=fallback_cache_duration)
```

Comment:
Could you write tests to make sure this expiry can be overridden by headers?

Basically, if a response contains `Cache-Control: no-cache` or `Cache-Control: max-age=31622400`, then the instruction from the header should be used instead of our override. I suspect that this heuristic currently overrides the header instead.

Reply:
I have added support for this and added tests.